File: Modulo 2/ClaseMapa/app/src/main/java/com/area51/clasemapa/MetodoRealmImpl.java
package com.area51.clasemapa;

import com.area51.clasemapa.realm.Ubicacion;

import java.util.UUID;

import io.realm.Realm;
import io.realm.RealmResults;

public class MetodoRealmImpl implements MetodoRealm {
    @Override
    public Ubicacion registrar(Ubicacion ubicacion) {
        Ubicacion resultado = null;
        Realm realm = Realm.getDefaultInstance();
        try {
            realm.beginTransaction();

            ubicacion.setId(UUID.randomUUID().toString());
            resultado = realm.copyToRealm(ubicacion);

            realm.commitTransaction();
        } catch (Exception e) {
            realm.cancelTransaction();
        }
        return resultado;
    }

    @Override
    public RealmResults<Ubicacion> obtener() {
        Realm realm = Realm.getDefaultInstance();
        return realm.where(Ubicacion.class)
                .findAll();
    }
}

File: Modulo 3/ClaseMVP/app/src/main/java/com/area51/clasemvp/net/database/MetodoRealmImpl.kt
package com.area51.clasemvp.net.database

import com.area51.clasemvp.net.entidad.Usuario
import io.realm.Realm

class MetodoRealmImpl : MetodoRealm {
    override fun agregar(usuario: Usuario): Usuario? {
        val realm = Realm.getDefaultInstance()
        var respuesta: Usuario? = null
        try {
            realm.beginTransaction()

            respuesta = realm.copyToRealm(usuario)

            realm.commitTransaction()
        } catch (e: Exception) {
            realm.cancelTransaction()
        }
        return respuesta
    }

    override fun validarUsuario(usuario: String, contrasena: String): Usuario? {
        val realm = Realm.getDefaultInstance()
        return realm.where(Usuario::class.java)
                .equalTo("usuario", usuario)
                .equalTo("contrasena", contrasena)
                .findFirst()
    }
}